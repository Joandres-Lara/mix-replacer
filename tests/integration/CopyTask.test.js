const fs = require("fs");
const path = require("path");
const Path = require("../../src/Path");
const CopyTask = require("../../src/CopyTask");
const {getTempDir,withHashQuery} = require("../helpers");

test("Normal flow", () => {
	const tmpdir = getTempDir();
	const src = "tests/integration/resources/browserconfig.xml";
	const target_dir = path.join(tmpdir, "non", "existing", "dir");
	const expected_target_path = path.join(target_dir, "browserconfig.xml");

	const task = new CopyTask(src, target_dir);
	const manifest = {
		"/images/mstile-150x150.png": withHashQuery("/images/mstile-150x150.png"),
		"/images/mstile-70x70.png": "/images/mstile-70x70.png",
		"/images/mstile-310x310.png": withHashQuery("/images/mstile-310x310.png"),
	};

	return task.run(manifest)
	           .then(target => {
		           expect(target).toBeInstanceOf(Path);

		           const target_path = target.toString();
		           expect(target_path).toBe(expected_target_path);

		           const target_content = fs.readFileSync(target_path).toString();
		           expect(target_content).toMatchSnapshot();
	           });
});

test("Throws Error", () => {
	const tmpdir = getTempDir();
	const src = "non/existent/file.txt";
	const target_dir = path.join(tmpdir, "non", "existing", "dir");
	const task = new CopyTask(src, target_dir);

	expect.assertions(2);

	return task.run({})
	           .catch(e => {
		           expect(e).toBeInstanceOf(Error);
		           expect(e.toString()).toMatch(/ENOENT/g);
	           });
});
