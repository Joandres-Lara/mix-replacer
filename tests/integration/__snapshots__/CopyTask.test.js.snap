// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normal flow 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<browserconfig>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png?id=5db8be17609750169201\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"/images/mstile-310x310.png\\"/>
            <TileColor>#70ba84</TileColor>
        </tile>
    </msapplication>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"/images/mstile-310x310.png\\"/>
            <TileColor>#471e7d</TileColor>
        </tile>
    </msapplication>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"/images/mstile-310x310.png?id=7b5572e4d23d5f53bf92\\"/>
            <TileColor>#ffffff</TileColor>
        </tile>
    </msapplication>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"/images/mstile-310x310.png\\"/>
            <TileColor>#ffffff</TileColor>
        </tile>
    </msapplication>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"/images/mstile-310x310.png\\"/>
            <TileColor>#e3bf86</TileColor>
        </tile>
    </msapplication>
    <msapplication>
        <tile>
            <square70x70logo src=\\"/images/mstile-70x70.png\\"/>
            <square150x150logo src=\\"/images/mstile-150x150.png?id=5db8be17609750169201\\"/>
            <wide310x150logo src=\\"/images/mstile-310x150.png\\"/>
            <square310x310logo src=\\"{{
            /images/mstile-310x310.png
            }}\\"/>
            <TileColor>#ffffff</TileColor>
        </tile>
    </msapplication>
</browserconfig>
"
`;
